presto-main/src/main/java/io/prestosql/server/remotetask/HttpRemoteTask.java

```diff
@@ -253,6 +259,8 @@ public HttpRemoteTask(
 
             taskStatusFetcher.addStateChangeListener(newStatus -> {
                 TaskState state = newStatus.getState();
+                newStatus.getDynamicTupleDomains()
+                        .forEach((key, value) -> dynamicFilterService.storeSummary(key, taskId, value));
```

Comment:
it seems for me that the dynamic filter integration is too low level here. I think `DynamicFilterService` should peek query info periodically (every `task.status-refresh-max-wait`) from `SqlQueryExecution` and build filters from it.
What do you think @martint ?

Reply:
Implemented polling of dynamic filters from `DynamicFilterService`